src/main/java/org/jabref/gui/libraryproperties/LibraryPropertiesDialogView.java

```diff
@@ -124,6 +129,13 @@ private void storeSettings() {
         Charset newEncoding = viewModel.selectedEncodingProperty().getValue();
         metaData.setEncoding(newEncoding);
 
+        BibDatabaseMode newMode = BibDatabaseMode.parse(viewModel.selectedDatabaseModeProperty().getValue());
+        BibDatabaseMode oldMode = metaData.getMode().orElse(null);
+        if (newMode != null && oldMode != newMode) {
```

Comment:
can you please move this code to `metadata.setMode`. Setting the mode to the same mode should not trigger any actions.